### PR TITLE
MULE-8430: Domain creates .mule folder on CWD instead of MULE_HOME

### PR DESCRIPTION
**Pull request by pablokraan**

### CODE REVIEW

File: modules/launcher/src/test/java/org/mule/module/launcher/domain/DomainMuleContextBuilderTestCase.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+@SmallTest
+public class DomainMuleContextBuilderTestCase extends AbstractMuleTestCase
+{
+
+    @Test
+    public void createsContainerConfiguration() throws Exception
+    {
+        DomainMuleContextBuilder builder = new DomainMuleContextBuilder("test");
+
```
Comment (marianogonzalez) [inline modules/launcher/src/test/java/org/mule/module/launcher/domain/DomainMuleContextBuilderTestCase.java]: remove empty spaces

File: modules/launcher/src/main/java/org/mule/module/launcher/domain/DomainMuleContextBuilder.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ public DomainMuleContextBuilder(String domainId)
     @Override
     protected MuleConfiguration getMuleConfiguration()
     {
-        DefaultMuleConfiguration defaultMuleConfiguration = new DefaultMuleConfiguration();
+        DefaultMuleConfiguration defaultMuleConfiguration = new DefaultMuleConfiguration(true);
```
Comment (marianogonzalez) [inline modules/launcher/src/main/java/org/mule/module/launcher/domain/DomainMuleContextBuilder.java]: if I migrate to a version containing this fix, does it mean that a new .mule will be created on MULE_HOME and whatever data stored in CWD will no longer be accessible? If so, shouldn't we include a migration guide entry saying that and instructing how to move the data over?

File: modules/launcher/src/test/java/org/mule/module/launcher/domain/DomainMuleContextBuilderTestCase.java (hunk 1 above)
Comment (pablokraan) [inline modules/launcher/src/test/java/org/mule/module/launcher/domain/DomainMuleContextBuilderTestCase.java]: what empty spaces?

Comment (marianogonzalez) [inline modules/launcher/src/test/java/org/mule/module/launcher/domain/DomainMuleContextBuilderTestCase.java]: meant empty lines between each test line

Comment (pablokraan) [inline modules/launcher/src/test/java/org/mule/module/launcher/domain/DomainMuleContextBuilderTestCase.java]: That empty lines are intended to separate the three conceptual parts of a test: setup, execute, assert

File: modules/launcher/src/main/java/org/mule/module/launcher/domain/DomainMuleContextBuilder.java (hunk 2 above)
Comment (pablokraan) [inline modules/launcher/src/main/java/org/mule/module/launcher/domain/DomainMuleContextBuilder.java]: Is a good point that this should be documented

**Pull request closed without merging**